simexpal/build.py

```diff
@@ -199,81 +199,89 @@ def do_step(step_yml, default_workdir=None):
-				stdout=subprocess.DEVNULL)
+				# Fetch the specified revision if it does not exist already.
+				verify_ref_result = subprocess.call(['git', '--git-dir', build.repo_dir,
+						'rev-parse', '-q', '--verify', generic_tag],
+					stdout=subprocess.DEVNULL)
 
-			if verify_ref_result != 0 or git_ref_changed():
-				# As we create generic_tag, we can add --no-tags here.
+				if verify_ref_result != 0 or git_ref_changed():
+					# As we create generic_tag, we can add --no-tags here.
+					subprocess.check_call(['git', '--git-dir', build.repo_dir,
+							'fetch', '--depth=1', '--no-tags',
+							build.info.git_repo] + fetch_refspec)
+
+				# Prune the existing worktree.
+				util.try_rmtree(build.clone_dir)
 				subprocess.check_call(['git', '--git-dir', build.repo_dir,
-						'fetch', '--depth=1', '--no-tags',
-						build.info.git_repo] + fetch_refspec)
-
-			# Prune the existing worktree.
-			util.try_rmtree(build.clone_dir)
-			subprocess.check_call(['git', '--git-dir', build.repo_dir,
-					'worktree', 'prune'])
-
-			# Recreate the worktree and check out the specified revision.
-			subprocess.check_call(['git', '--git-dir', build.repo_dir,
-					'worktree', 'add', '--detach',
-					build.clone_dir,
-					generic_tag])
-		else:
-			# Recreate the source directory
-			util.try_rmtree(build.source_dir)
-			util.try_mkdir(build.source_dir)
+						'worktree', 'prune'])
 
-			# Clone the git repository into the build.source_dir
-			subprocess.check_call(['git', 'clone', build.info.git_repo, build.source_dir])
+				# Recreate the worktree and check out the specified revision.
+				subprocess.check_call(['git', '--git-dir', build.repo_dir,
+						'worktree', 'add', '--detach',
+						build.clone_dir,
+						generic_tag])
+			else:
+				# Recreate the source directory
+				util.try_rmtree(build.source_dir)
+				util.try_mkdir(build.source_dir)
 
-		util.touch(os.path.join(checkout_dir, 'checkedout.simexpal'))
+				# Clone the git repository into the build.source_dir
+				subprocess.check_call(['git', 'clone', build.info.git_repo, build.source_dir])
 
-		if build.info.recursive_clone:
-			# Clone submodules recursively
-			subprocess.check_call(['git', 'submodule',
-					'update', '--init', '--recursive'], cwd=checkout_dir)
+			util.touch(os.path.join(checkout_dir, 'checkedout.simexpal'))
 
-		did_work = True
+			if build.info.recursive_clone:
+				# Clone submodules recursively
+				subprocess.check_call(['git', 'submodule',
+						'update', '--init', '--recursive'], cwd=checkout_dir)
+
+			did_work = True
+		else:
+			print("simexpal: Skipping checkout-phase for git-less dev-build {}".format(build.name))
+
+	# Make sure that the build source directory exists for git-less dev-builds.
+	if build.info.git_repo is None and build.revision.is_dev_build and not os.path.isdir(build.source_dir):
+		raise RuntimeError(f"The source directory of the build '{build.name}' with revision '{build.revision.name}'"
+							f" does not exist.")
```

Comment:
Why do we check for `build.revision.is_dev_build` here?

Reply:
``build.source_dir`` is only available to dev-builds so that we have to do this additional check